internal/app/connectconformance/results.go

```diff
@@ -547,3 +535,118 @@ func indent(s string) string {
 	}
 	return strings.Join(lines, "\n")
 }
+
+func checkWireDetails(testCaseReq *conformancev1.ClientCompatRequest, expected, actual *conformancev1.ClientResponseResult) multiErrors {
+	expectedWire := expected.WireDetails
+	actualWire := actual.WireDetails
+	if actualWire == nil {
+		return nil
+	}
+	var errs multiErrors
+
+	if expectedWire.GetActualStatusCode() != 0 {
+		if diff := cmp.Diff(expectedWire.ActualStatusCode, actualWire.ActualStatusCode, protocmp.Transform()); diff != "" {
+			errs = append(errs, fmt.Errorf("actual HTTP status code does not match: - wanted, + got\n%s", diff))
+		}
+	}
+
+	switch testCaseReq.Protocol {
+	case conformancev1.Protocol_PROTOCOL_CONNECT:
+		if len(actualWire.ActualHttpTrailers) > 0 {
+			errs = append(errs, fmt.Errorf("connect response included %d HTTP trailers but should not have any", len(actualWire.ActualHttpTrailers)))
+		}
+
+		// TODO - Add comparison (and tests) for connecterrorraw
+
+		// if diff := cmp.Diff(expectedWire.ConnectErrorRaw, actualWire.ConnectErrorRaw, protocmp.Transform()); diff != "" {
+		// 	errs = append(errs, fmt.Errorf("raw Connect error does not match: - wanted, + got\n%s", diff))
+		// }
+
+	case conformancev1.Protocol_PROTOCOL_GRPC:
+		if len(actualWire.ActualHttpTrailers) > 0 {
+			// make sure on-the-wire trailers are a super-set of reported trailers
+			return append(errs, checkHeaders("reported trailers vs raw HTTP trailers", actual.ResponseTrailers, actualWire.ActualHttpTrailers)...)
+		}
+		// We might also check that there *are* actual HTTP trailers if the number of
+		// actual payloads is non-zero, since gRPC protocol requires them if there's
+		// any data in the body. But the reference client implementation already
+		// handles that part of the spec and returns an UNKNOWN error when that happens.
+		//
+		// Another assertion that would be nice to do is that trailers-only responses
+		// actually set the "end stream" bit on the headers HTTP/2 frame, to disallow
+		// an empty HTTP/2 data frame with this style of response. This is part of the
+		// gRPC spec. But "net/http" doesn't provide that level of visibility into the
+		// underlying HTTP/2 stream frames. Luckily, the grpc-go reference client DOES
+		// check this. So we can still catch this condition in a misbehaving server.
+
+	case conformancev1.Protocol_PROTOCOL_GRPC_WEB:
+		if len(actualWire.ActualHttpTrailers) > 0 {
+			errs = append(errs, fmt.Errorf("grpc-web response included %d HTTP trailers but should not have any", len(actualWire.ActualHttpTrailers)))
+		}
+		if actualWire.ActualGrpcwebTrailers != nil {
+			errs = append(errs, checkGRPCWebTrailers(actualWire.GetActualGrpcwebTrailers(), actual.ResponseTrailers)...)
```

Comment:
Does any of the validation being done in `checkGRPCWebTrailers` apply to gRPC trailers also? I know there are some differences, but was just curious why we validate them for gRPC-web but don't validate anything for gRPC.

Reply:
For gRPC, "net/http" already does all of the validation necessary. They are actual HTTP trailers, and "net/http" is responsible for processing them and providing them in the form of a `http.Header` on the response object.

For gRPC-web, this data is in a chunk of the body, so the application has to process it. And connect-go is a little lenient for these -- which is fine (even preferable) to adhere to the robustness principle -- "be liberal about what you accept but conservative about what you produce". So these are extra checks, just to be more persnickety about the on-the-wire form of gRPC-web trailers.